### DISCUSSION TITLE
ESP32-S2-LCD-0.96

### DISCUSSION THREAD

**Discussion by kjm1102** [micropython/micropython#14234] [Libraries & Drivers]
I'm trying to get the 0.96" 7735 LCD on the https://www.waveshare.com/wiki/ESP32-S2-Pico board working with upython. I got the ST7735.py and sysfont.py at https://github.com/cheungbx/st7735-esp8266-micropython/blob/master/st7735.py loaded but when I run test80x160.py the LCD remains blank.

I think the lines
spi = SPI(1, baudrate=8000000, polarity=0, phase=0)
tft=TFT(spi,16)
probably need editing with new pin numbers but I can't figure out from the LCD USED PINS box at https://www.waveshare.com/wiki/File:ESP32-S2-Pico-details-9-2.jpg what they should be.

Anyone got any suggestions for navigating the inscrutable?

**Comment by shariltumin** [comment]
This is what you can do to find out which PINs are used for your esp32s2 board.
For example, my board:

```bash
>>>
>>> 
MPY: soft reboot
MicroPython v1.20.0-206-g33b403dfb on 2023-10-29; ESP32 S2_MINI WEMOS with ESP32-S2FN4R2
>>> from machine import SPI
>>> spi=SPI(1, baudrate=8000000, polarity=0, phase=0)
>>> spi
SPI(id=1, baudrate=8000000, polarity=0, phase=0, bits=8, firstbit=0, sck=7, mosi=9, miso=11)
>>>
```

Hope this helps.

**Reply by kjm1102** [reply]
Tnx for the heads up Sharil. I added the waveshare pinouts
```
spi = SPI(1, baudrate=8000000, polarity=0, phase=0, sck=10, mosi=11)
dc=Pin(18, Pin.OUT); rst=Pin(21, Pin.OUT); cs=Pin(9, Pin.OUT); bl=Pin(45, Pin.OUT); bl(1)
tft=TFT(spi,dc,rst,cs,80,160)
```
to the least badly documented 7735 lib I could find online & got a result. It ain't pretty but it's a start!
![2404ESP32-S2-LCD-0 96](https://github.com/micropython/micropython/assets/92007731/073a95f0-33e2-4412-899b-fb34f41e8b3a)
